libasn1print/asn1printproto.c

```diff
@@ -318,9 +437,22 @@ asn1print_expr_proto(asn1p_t *asn, asn1p_module_t *mod, asn1p_expr_t *expr,
 		}
 
 		proto_msg_def_t *msgelem = proto_create_msg_elem("value", "int32", NULL);
+
 		switch (expr->expr_type) {
 		case ASN_BASIC_INTEGER:
 			if (expr->constraints != NULL) {
+                // adding APER tags
+                long lowerbound;
+                lowerbound = get_lowerbound(expr->constraints);
+                if(lowerbound != -1) {
+                    msgelem->tags.valueLB = lowerbound;
+                }
+                // obtaining upperbound
+                long upperbound;
+                upperbound = get_upperbound(expr->constraints);
+                if(upperbound != -1) {
+                    msgelem->tags.valueUB = upperbound;
+                }
 				char *constraints = proto_constraint_print(expr->constraints, flags | APF_INT32_VALUE);
```

Comment:
I'm not sure your approach will cover all circumstances here.
`proto_constraint_print()` is already getting these values from constraints. 
OK - it is returning a string, but maybe you could pass it the `msgelem` and add to it's tags inside the function.
I know it's horrible and recursive and all that, but that's because Constraints are horrible and recursive too.